src/Build/BackEnd/Components/RequestBuilder/Lookup.cs

```diff
@@ -663,12 +663,32 @@ internal void AddNewItemsOfItemType(string itemType, ICollection<ProjectItemInst
 
                 // Ensure we don't also add any that already exist.
                 var existingItems = GetItems(itemType);
+              
+                var existingItemsHashSet = existingItems.ToHashSet(ProjectItemInstance.EqualityComparer);
+                
                 if (existingItems.Count > 0)
                 {
-                    itemsToAdd = itemsToAdd.Where(item => !existingItems.Contains(item, ProjectItemInstance.EqualityComparer));
+                    
+                    var deduplicatedItemsToAdd = new List<ProjectItemInstance>();
+                    foreach (var item in itemsToAdd) {
+                        if (!existingItemsHashSet.Contains(item)) {
+                            deduplicatedItemsToAdd.Add(item);
```

Comment:
this no longer deduplicates if the itemsToAdd has duplicates.

Instead, you can do:
```csharp
if (existingItemsHasSet.Add(item))
{
    deduplicatedItemsToAdd.Add(item);
}
```

Reply:
HashSet.Add returns false if the item was already there

Reply:
There is still this few lines above so the itemsToAdd has no duplicates.
`itemsToAdd = itemsToAdd.Distinct(ProjectItemInstance.EqualityComparer);`
However your construct looks cleaner. Thanks for the suggestion.

Reply:
With your construct we can skip the Distinct altogether, saving one extra pass over the collection I think.

Reply:
Still needed if the list is empty, in the "compute everything case" I removed it and used your construct.